packages/webpack/src/config/server.js

```diff
@@ -20,11 +20,23 @@ export default class WebpackServerConfig extends WebpackBaseConfig {
 
   get externalsWhitelist () {
     return [
-      /\.(?!js(x|on)?$)/i,
+      this.isNonNativeImport.bind(this),
       ...this.normalizeTranspile()
     ]
   }
 
+  /**
+   * files *not* ending on js|jsx|json should be processed by webpack
+   *
+   * this might (still) generate false-positives for
+   * - someFile.umd (referring to someFile.umd.js)
+   * - some.folder (some.folder being a folder containing a package.json)
+   */
+  isNonNativeImport (modulePath) {
+    const extname = path.extname(modulePath)
+    return extname !== '' && !/^\.js(x|on)?$/.test(extname)
```

Comment:
This would introduce more edge cases as name.[type].[ext] is not uncommon for node_modules...

What if we remove ^ from regex and simply test against path.basename? Linked issue is for dot inside part of path so i think it would still fix that ☺️

Reply:
@pi0 thanks for the amazingly fast response. I might be missing something but would that change the behaviour?

```javascript
/^\.js(x|on)?$/.test(path.extname('file.json.md'))  // false
/^\.js(x|on)?$/.test(path.extname('file.json'))        // true

/\.js(x|on)?$/.test(path.basename('file.json.md')) // false
/\.js(x|on)?$/.test(path.basename('file.json'))        // true

since
path.extname('file.json.md')                                     // '.md'
path.extname('file.json')                                           // '.json'
```

Reply:
I mean $ should be still there 😅 But not restriction for start to allow patterns like entry.min.js 

Not related but I'm wondering why jsx is considered native? It should be only js/json (aso mjs) i guess

Reply:
I also wondered about `jsx` but didn't want to change more than necessary. I can however remove jsx and allow mjs if you'd like.

About extname / basename:
```
path.extname('entry.min.js') === path.extname('entry.js')
```

Therefore patterns like `name.[type].[ext]` should not make any difference (node 12.16.1).

Reply:
I adjusted the comment to make it (hopefully) more clear.

IMHO the whitelisting should be removed completely and left to the user, due to the limitations of the implementation. But that might be a breaking change...

Reply:
I agree. That's why we support `--standalone` and `options.build.standalone`